### PR TITLE
[react-select] Allow more types for value prop

### PR DESCRIPTION
**Pull request by Methuselah96**
Many of the tests in https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/__tests__/Select.test.js assume that the type of the value doesn't have to correspond with `isMulti`.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: N/A
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-select/src/Select.d.ts [hunk 1]
```diff
@@ -188,11 +188,11 @@ export interface NamedProps<
     /** Select the currently focused option when the user presses tab */
     tabSelectsValue?: boolean;
     /** The value of the select; reflected by the selected option */
-    value?: ValueType<OptionType, IsMulti>;
+    value?: readonly OptionType[] | OptionType | null;
```
[the remark this answers is not in the document]
Comment (kylemh) [reply types/react-select/src/Select.d.ts:191]: Also, it seems like this type is leveraged really often. 

Could be useful to do:

```ts
type ValueType<OptionType = OptionTypeBase> = readonly OptionType[] | OptionType | null;
```

for the sake of DRY 🤷‍♂️ 

Not a showstopper though, feel free to merge anyways.

**Review by kylemh — APPROVED** [review]

File: types/react-select/src/Select.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Methuselah96) [reply types/react-select/src/Select.d.ts:190]: Yeah, a type could be useful, but `ValueType` is already taken. I'll try to think of a good name.

[the remark this answers is not in the document]
Comment (kylemh) [reply types/react-select/src/Select.d.ts:190]: `ReactSelectValueType` ?

It's lengthy, but unarguably correct 😂

[the remark this answers is not in the document]
Comment (Methuselah96) [reply types/react-select/src/Select.d.ts:190]: The existing `ValueType` is already for react-select, so I don't think it provides a helpful distinction. Can we go ahead and merge this? I'll make change the existing `ValueType` to `OnChangeValueType` in a follow-up PR.

[the remark this answers is not in the document]
Comment (kylemh) [reply types/react-select/src/Select.d.ts:190]: Sure! I already approved it and said this was optional!

[the remark this answers is not in the document]
Comment (Methuselah96) [reply types/react-select/src/Select.d.ts:190]: Sorry for the confusion! I think you need to re-approve it because I had to resolve a merge conflict.

**Review by kylemh — APPROVED** [review]

**Comment by Methuselah96** [comment]
Ready to merge

**Pull request merged**